### PR TITLE
Descending Keys Tests

### PR DESCRIPTION
**Pull request by andytill**
Tests for descending keys.

### CODE REVIEW

File: tests/ts_updown_select_SUITE.erl [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+-module(ts_updown_select_SUITE).
```
Comment (hazen) [inline tests/ts_updown_select_SUITE.erl:20]: Since this is a cluster, I would suggest `ts_cluster_updown_select_SUITE`

**Comment by gordonguthrie** [comment]
+1 6681d5d

**Comment by hazen** [comment]
@andytill Why do we have so many changes on this?  Was this some very old version which you started from? Maybe we could rebase your changes?

File: run_tests.sh [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+./riak_test -c rt_ts -t tests/ts_updown_select_SUITE.erl
```
Comment (hazen) [inline run_tests.sh:2]: This seems pretty specific.  Was this meant to be included in this PR?

File: src/riak_test_escript.erl [hunk 3]
```diff
@@ -149,6 +149,7 @@ main(Args) ->
                     Tests0;
                 {_, undefined} ->
                     Tests0;
+                %% TODO don't understand what this does
```
Comment (hazen) [inline src/riak_test_escript.erl:151]: Did you want to add this comment?  I can explain (more) of what this is meant to do.  When `riak_test` was written it was supposed to run a whole bunch of tests sequentially and this shuffles the order.  Not that useful now that we have independent tests on giddyup.  It was meant to break any potential test interdependence.

File: tests/ts_cluster_ddl_upgrades_SUITE.erl [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+%% git clone git@github.com:basho/riak.git && git checkout riak_ts-1.4.0 && make locked-deps
+-define(TS_VERSION_1_4, riak_ts_1_4).
+
+-define(DDL_REC_CAP, {riak_kv, riak_ql_ddl_rec_version}).
+
+suite() ->
+    [{timetrap,{minutes,10}}].
+
+init_per_suite(Config) ->
+    % ct:pal("VERSION 1.4 ~p", [rt:find_version_by_name(["riak_ts-1.4.0", "riak_ts_ee-1.4.0"])]),
```
Comment (hazen) [inline tests/ts_cluster_ddl_upgrades_SUITE.erl:43]: Mean to leave this in?

File: tests/ts_updown_select_SUITE.erl (hunk 1 above)
Comment (hazen) [reply tests/ts_updown_select_SUITE.erl:20]: Yup. Still think that's a more consistent name

**Review by hazen — CHANGES_REQUESTED** [review]
Let me know your thoughts on my comments. Thanks!

File: src/riak_test_escript.erl (hunk 3 above)
Comment (andytill) [reply src/riak_test_escript.erl:151]: Not sure where that is from. I need to rebase this, it will probably disappear then.

Comment (gordonguthrie) [reply src/riak_test_escript.erl:151]: I left it in there - it was something to do with parallel tests running in the cloud - but I don't understand why/how it is used...

**Comment by hazen** [comment]
Think we'll need a rebase on this PR since I doubt we actually have 103 relevant file changes...

**Comment by andytill** [comment]
@javajolt I will close this one and open another PR tomorrow, something went very wrong with the rebase I did today.

**Comment by andytill** [comment]
Closing and reopening as https://github.com/basho/riak_test/pull/1200

**Pull request closed without merging**